jobs/build/promote-assembly/Jenkinsfile

```diff
@@ -0,0 +1,358 @@
+                    ),
+                    booleanParam(
+                        name: 'SKIP_IMAGE_LIST',
+                        description: '(Standard Release) Do not gather an advisory image list for docs.',
+                        defaultValue: false,
+                    ),
+                    booleanParam(
+                        name: 'SKIP_SIGNING',
+                        description: 'Do not sign the release.',
+                        defaultValue: false,
+                    ),
+                    booleanParam(
+                        name: 'FORCE_RHCOS_SYNC',
+                        description: '(Standard Release) Always sync RHCOS images to mirrors.',
+                        defaultValue: false,
+                    ),
+                    string(
+                        name: 'MAIL_LIST_SUCCESS',
+                        description: 'Success Mailing List',
+                        defaultValue: [
+                            'aos-cicd@redhat.com',
+                            'aos-qe@redhat.com',
+                            'aos-art-automation+new-release@redhat.com',
+                        ].join(','),
+                        trim: true,
+                    ),
+                    commonlib.dryrunParam('Take no actions. Note: still notifies and runs signing job (which fails)'),
+                    commonlib.mockParam(),
+                ]
+            ],
+        ]
+    )
+
+    commonlib.checkMock()
+    def (major, minor) = commonlib.extractMajorMinorVersionNumbers(params.VERSION)
+    // TODO: Move commonlib.ocpReleaseState to ocp-build-data or somewhere.
+    def arches = params.ARCHES? commonlib.parseList(params.ARCHES) : commonlib.ocpReleaseState["${major}.${minor}"]?.release
+    if (!arches) {
+        error("Could determine which architectures to release. Make sure they are specified in `commonlib.ocpReleaseState`.")
+    }
+    def release_offset = params.RELEASE_OFFSET? Integer.parseInt(params.RELEASE_OFFSET) : null
+    def skipAttachedBugCheck = false
+    def next_minor = "${major}.${minor + 1}"
+    if (!(commonlib.ocpReleaseState[next_minor] && commonlib.ocpReleaseState[next_minor]["release"])) {
+        // If next_minor is not released yet, don't check attatched bugs.
+        // TODO: CVE flaw check should be exempted after https://github.com/openshift/elliott/pull/239 is merged.
+        skipAttachedBugCheck = true
+    }
+    if (params.DRY_RUN) {
+        currentBuild.displayName += " (dry-run)"
+        currentBuild.description += " [DRY RUN]"
+    }
+    currentBuild.displayName += " ${params.VERSION} - ${params.ASSEMBLY}"
+
+    stage("Initialize") {
+        // Install pyartcd
+        commonlib.shell(script: "pip install -e ./pyartcd")
+        // must be able to access remote registry for verification
+        buildlib.registry_quay_dev_login()
+    }
+
+    def release_info = {}
+
+    stage("promote release") {
+        sh "mkdir -p ./artcd_working"
```

Comment:
As a defensive action, could a `rm -rf ./artcd_working` be executed?

Reply:
Yes, that's a good idea.